### ISSUE TITLE
discussion: revamp (+fix) 2200s

### ISSUE THREAD

**Issue by Thorin-Oakenpants** [ghacksuserjs/ghacks-user.js#371]
> 2204 requires browser.link.open_newwindow=3

I removed 2204 in https://github.com/ghacksuserjs/ghacks-user.js/commit/40da3653d36c72686b90a1dfbe46b4d61b524ae0

for the record to stop looking it up, it was
```js
/* 5007: open new windows in a new tab instead
 * 1=current window, 2=new window, 3=most recent window
 * [SETTING] Options>General>Tabs>Open new windows in a new tab instead ***/
 user_pref("browser.link.open_newwindow", 3);
```

and something about `browser.link.open_newwindow.restriction` being added (to the 2200's?)

**Comment by earthlng** [comment]
> /* 2204: ...
> [TEST] https://people.torproject.org/~gk/misc/entire_desktop.html

set browser.link.open_newwindow to 2 and see what happens

**Comment by Thorin-Oakenpants** [comment]
Thanks. That's interesting - a value of 1 seems to block the whole thing

**Comment by earthlng** [comment]
a value of 1 for which one?

**Comment by Thorin-Oakenpants** [comment]
Do you think we should move 2418 -> 2200's as well? So something like

2203: no changes
2204: `browser.link.open_newwindow=2`
2205: `browser.link.open_newwindow.restriction` (new)
2206: `full-screen-api.enabled=false` (was 2418)
2207: the current 2203 with the info "requires that one 2203 entry, 2205, 2206"
2208: `dom.disable_beforeunload` (was 2205)

------
> a value of 1 for which one?

> /* 2204: ...
> [TEST] https://people.torproject.org/~gk/misc/entire_desktop.html

set browser.link.open_newwindow to 1 and see what happens (as per the commit linked in first post, the UI checkbox is boolean, I just thought I would test it with 1, maybe that value is undocumented/obsolete)

**Comment by earthlng** [comment]
with value 1 it opens the "new window" in the current tab.
yes 2418 would probably fit better in 2200.

IDK why 2202 + 2203 are separated like that. All the `dom.disable_window_open_feature.*` should be grouped together because those are window.open() "features". The other 4 prefs in 2203 meddle with already open windows.

`browser.link.open_newwindow.restriction=0` + `browser.link.open_newwindow=3` makes all the `disable_window_open_feature` prefs obsolete anyway but I guess there's no harm in keeping them

**Comment by earthlng** [comment]
/* 2418: disable full-screen API
 * false=block, true=ask ***/

there's no asking: https://developer.mozilla.org/samples/domref/fullscreen.html

**Comment by earthlng** [comment]
actually 2418 is more about not leaking the real monitor resolution than UI meddling. 2205 isn't really UI meddling either.

**Comment by Thorin-Oakenpants** [comment]
also worth noting that RFP covers new window sizes (ticket was closed as a duplicate), but I did not move these 2200 prefs to RFP Alt because new window blocking is still a bloody good idea / feature IMO. The line `this is to stop malicious window sizes and screen res leaks` can be cleaned up to reflect this. "meddling" might not be the best word to use, but I def think the window resizing stuff fits.

AFAIK, the full-screen api applies to websites ability to make you go full screen, but I'm not sure - eg a youtube video button that requires user interaction, or it can be automagic. There doesn't seem to be a Moz prompt in site permissions - as you say, no asking. Of note, RFP is meant to be bringing in a prompt for maximizing - I wonder if they will do the same for full-screen. User's can still go full-screen manually.

All up I think all these belong in this section, ad with the new additions etc its basically a revamp. I'll try and get something done soon for preview

**Comment by Thorin-Oakenpants** [comment]
from #375 billboard1: `dom.disable_open_during_load` => about:preferences#privacy > Permissions > Block popup windows. Default is true (which is what we want) - this would fit under the 2200's which needs a bit of a revamp, IF we want to include it.

**Comment by Thorin-Oakenpants** [comment]
Note: popups stuff, we could also move 2414 + 2414b (both popup events) into the 2200's

**Comment by earthlng** [comment]
for starters lets re-group 2202 + 2203 and move the link from the header to 2202:

```js
/*** 2200: UI MEDDLING ***/
/* 2202: prevent websites from disabling new window features
 * [1] http://kb.mozillazine.org/Prevent_websites_from_disabling_new_window_features ***/
user_pref("dom.disable_window_open_feature.close", true);
user_pref("dom.disable_window_open_feature.location", true); // default: true
user_pref("dom.disable_window_open_feature.menubar", true);
user_pref("dom.disable_window_open_feature.minimizable", true);
user_pref("dom.disable_window_open_feature.personalbar", true); // bookmarks toolbar
user_pref("dom.disable_window_open_feature.resizable", true); // default: true
user_pref("dom.disable_window_open_feature.status", true); // default: true
user_pref("dom.disable_window_open_feature.titlebar", true);
user_pref("dom.disable_window_open_feature.toolbar", true);
/* 2203: disable meddling with open windows ***/
user_pref("dom.allow_scripts_to_close_windows", false); // default: false
user_pref("dom.disable_window_flip", true); // window z-order (default: true)
user_pref("dom.disable_window_move_resize", true);
```

remove the 3 `// default: true` in 2202 if you want.
In 2203 `allow_scripts_to_close_windows` is default false and `disable_window_flip` default true so IDK if we really need those at all.

**Comment by earthlng** [comment]
> RFP covers new window sizes

oh I didn't know that, I only tested in ESR.
In ESR browser.link.open_newwindow=2 always leaks regardless of .open_newwindow.restriction.
browser.link.open_newwindow=3 leaks when .restriction is anything but 0.
So only in ESR does 2204 still require browser.link.open_newwindow=3 to stop screen-res leaks.

However in non-ESR the [testpage](https://people.torproject.org/~gk/misc/entire_desktop.html) shows an unexpected result of 10x10 with these settings. Maybe best to put these 2 together and move them under RFP ALTS?

**Comment by Thorin-Oakenpants** [comment]
I don't want to put anything under RFP ALT, because these are good non-window-fuckery items regardless of the FP aspect. I have a kind of draft new section here: the ref to mozillazine got trashed (we should not use that site, or at least links that have outdated info) and I renamed the section as `/*** 2200: WINDOW & UI MEDDLING ***/`, pulled in the two popup prefs from 2400s, added dom.popup_allowed_events (new) and also pulled in `full-screen-api.enabled` from 2400s. I was just messing around with it, the order (and will work in the RFP info)

user_pref("browser.link.open_newwindow", 3); << this must be 3 right? i.e for browser.link.open_newwindow.restriction=0 to work? Just want to clarify that

**Comment by Thorin-Oakenpants** [comment]
kinda OT: are we going to treat ESR60 as the trigger point (eg removing ESR settings info, new ESR deprecated section, etc), or wait for EOL of 52.9?

When are u planning on changing - you're missing out on a lot of improvements, and I don't mean the speed stuff? Some stuff I don't think will be ready 100% until 62+ anyway (eg storage, so many quirks with extensions). At least you didn't have to go thru any of the pains on the train ride, but you will have a bit of a curve. At least my strategy of default deny all storage worked in my favor - I feel sorry for all the mugs who used cookie extensions (that left behind orphaned data and gave them away all the time).

**Comment by earthlng** [comment]
I'll wait for EOL

**Comment by Thorin-Oakenpants** [comment]
```
/* 2418: disable Fullscreen API
 * [TEST] https://developer.mozilla.org/samples/domref/fullscreen.html ***/
user_pref("full-screen-api.enabled", false);
```

This is problematic. While users can use F11 to toggle fullscreen, that's the page content, not the video. In the test, you can either use press the Return or Enter key anywhere in the page, OR click the bottom right icon on the video nav bar. With the pref at false, there is no way to get the video-only at fullscreen

Not that I use them, but doesn't this mess with `user-initiated` fullscreen on youtube, netflix, vimeo, twitch etc?

**Comment by Thorin-Oakenpants** [comment]
ignore the number (I'm still reworking the 2200s), does this look OK?

```js
/* 2418: disable Fullscreen API [SETUP]
 * [NOTE] You can still manually toggle the browser's fullscreen state (F11), 
 * but this will disable embedded video/game fullscreen controls, e.g. youtube
 * [TEST] https://developer.mozilla.org/samples/domref/fullscreen.html ***/
user_pref("full-screen-api.enabled", false);
```

Note: its `fullscreen` (one word) - see https://developer.mozilla.org/en-US/docs/Web/API/Fullscreen_API

**Comment by Thorin-Oakenpants** [comment]
I'll post here, so its easier to read

```js
/*** 2200: UI / WINDOWS / POPUPS ***/
user_pref("_user.js.parrot", "2200 syntax error: the parrot's 'istory!");
/* 2201: disable website control over browser right-click context menu
 * [NOTE] Shift-Right-Click will always bring up the browser right-click context menu ***/
   // user_pref("dom.event.contextmenu.enabled", false);
/* 2202: disable [new window] scripts hiding or disabling the following ***/
user_pref("dom.disable_window_open_feature.location", true);
user_pref("dom.disable_window_open_feature.menubar", true);
user_pref("dom.disable_window_open_feature.resizable", true);
user_pref("dom.disable_window_open_feature.status", true);
user_pref("dom.disable_window_open_feature.toolbar", true);
/* 2203: disable [popup window] scripts hiding or disabling the following ***/
user_pref("dom.disable_window_flip", true); // window z-order
user_pref("dom.disable_window_move_resize", true);
user_pref("dom.disable_window_open_feature.close", true);
user_pref("dom.disable_window_open_feature.minimizable", true);
user_pref("dom.disable_window_open_feature.personalbar", true); // bookmarks toolbar
user_pref("dom.disable_window_open_feature.titlebar", true);
user_pref("dom.allow_scripts_to_close_windows", false);
/* 2204: disable links opening in a new window
 * This is to stop malicious window sizes and screen res leaks etc in conjunction with
 * 2203's dom.disable_window_move_resize=true and 2206's full-screen-api.enabled=false
 * [NOTE] ESR52 and lower requires 2205's browser.link.open_newwindow=3
 * [NOTE] RFP (4500) already resizes new windows to cover screen res leaks
 * [NOTE] You can still right click a link and select open in a new window
 * [TEST] https://people.torproject.org/~gk/misc/entire_desktop.html
 * [1] https://trac.torproject.org/projects/tor/ticket/9881 ***/
user_pref("browser.link.open_newwindow.restriction", 0);
/* 2205: open new windows in a new tab instead
 * 1=current window, 2=new window, 3=most recent window
 * [SETTING] Options>General>Tabs>Open new windows in a new tab instead ***/
user_pref("browser.link.open_newwindow", 3);
/* 2206: disable Fullscreen API [SETUP]
 * [NOTE] You can still manually toggle the browser's fullscreen state (F11), 
 * but this will disable embedded video/game fullscreen controls, e.g. youtube
 * [TEST] https://developer.mozilla.org/samples/domref/fullscreen.html ***/
user_pref("full-screen-api.enabled", false);
/* 2210: block popup windows
 * [SETTING] Options>Privacy & Security>Permissions>Block pop-up windows ***/
user_pref("dom.disable_open_during_load", true);
/* 2211: set max popups from a single non-click event - default is 20! ***/
user_pref("dom.popup_maximum", 3);
/* 2212: limit events that can cause a popup
 * default is "change click dblclick mouseup pointerup notificationclick reset submit touchend"
 * [1] http://kb.mozillazine.org/Dom.popup_allowed_events ***/
user_pref("dom.popup_allowed_events", "click dblclick");
/* 2213: disable "Confirm you want to leave" dialog on page close
 * Does not prevent JS leaks of the page close event.
 * [1] https://developer.mozilla.org/docs/Web/Events/beforeunload
 * [2] https://support.mozilla.org/questions/1043508 ***/
user_pref("dom.disable_beforeunload", true);
```

- `2213` doesn't really belong - maybe move that to 2400s :question: 
- `2206` (fullscreen) was `2418`
- `2207` is new
- `2211 + 2212` (popup stuff) were `2415 + 2415b` respectively

**Comment by earthlng** [comment]
simply "UI" without meddling or so is confusing IMO, I would expect more things to be in this section if it's about "UI".
I would title it something like "Window meddling and leaks & Popups" and move 2201 + 2213 to 2400 and re-group 2202+2203 the way I said.
2204+2205: 2205 is the main pref; 2204=0 means "divert all links/windows opened with window.open() according to 2205" so IMO 2205 should come first and the connection between these 2 should be made more clear. The note about right-click would probably fit better under 2205, too.

**Comment by Thorin-Oakenpants** [comment]
- moving context menu out would make it easier, agreed - `DONE`
- I removed the word meddling because I thought you didn't like it - twas just a suggestion - `DONE`
- 2004+2205 : i can revere the order, no big deal - `DONE`

---
- re-group 2202+2203 the way u said? How was that again? If I remember correctly you removed some(?) and mixed them up. I like them split by `new window` and `popup`. Refresh my memory, please
- " the connection between these 2 should be made more clear." - can you elaborate?
- `[NOTE] ESR52 and lower requires 2203's browser.link.open_newwindow=3` is not the whole truth, in stable it still can't be `2`, so we should just leave simplify it as all releases require `3` and be done with it - IDK

---
here's a new one since the numbers changed (note two items will be moved to 2400s and not included)
```js
/*** 2200: WINDOW MEDDLING/LEAKS & POPUPS ***/
user_pref("_user.js.parrot", "2200 syntax error: the parrot's 'istory!");
/* 2201: disable [new window] scripts hiding or disabling the following ***/
user_pref("dom.disable_window_open_feature.location", true);
user_pref("dom.disable_window_open_feature.menubar", true);
user_pref("dom.disable_window_open_feature.resizable", true);
user_pref("dom.disable_window_open_feature.status", true);
user_pref("dom.disable_window_open_feature.toolbar", true);
/* 2202: disable [popup window] scripts hiding or disabling the following ***/
user_pref("dom.disable_window_flip", true); // window z-order
user_pref("dom.disable_window_move_resize", true);
user_pref("dom.disable_window_open_feature.close", true);
user_pref("dom.disable_window_open_feature.minimizable", true);
user_pref("dom.disable_window_open_feature.personalbar", true); // bookmarks toolbar
user_pref("dom.disable_window_open_feature.titlebar", true);
user_pref("dom.allow_scripts_to_close_windows", false);
/* 2203: open new windows in a new tab instead
 * 1=current window, 2=new window, 3=most recent window
 * [SETTING] Options>General>Tabs>Open new windows in a new tab instead ***/
user_pref("browser.link.open_newwindow", 3);
/* 2204: disable links opening in a new window
 * This is to stop malicious window sizes and screen res leaks etc in conjunction with
 * 2202's dom.disable_window_move_resize=true and 2205's full-screen-api.enabled=false
 * [NOTE] ESR52 and lower requires 2203's browser.link.open_newwindow=3
 * [NOTE] RFP (4500) already resizes new windows to cover screen res leaks
 * [NOTE] You can still right click a link and select open in a new window
 * [TEST] https://people.torproject.org/~gk/misc/entire_desktop.html
 * [1] https://trac.torproject.org/projects/tor/ticket/9881 ***/
user_pref("browser.link.open_newwindow.restriction", 0);
/* 2205: disable Fullscreen API [SETUP]
 * [NOTE] You can still manually toggle the browser's fullscreen state (F11), 
 * but this will disable embedded video/game fullscreen controls, e.g. youtube
 * [TEST] https://developer.mozilla.org/samples/domref/fullscreen.html ***/
user_pref("full-screen-api.enabled", false);
/* 2206: block popup windows
 * [SETTING] Options>Privacy & Security>Permissions>Block pop-up windows ***/
user_pref("dom.disable_open_during_load", true);
/* 2207 set max popups from a single non-click event - default is 20! ***/
user_pref("dom.popup_maximum", 3);
/* 2208: limit events that can cause a popup
 * default is "change click dblclick mouseup pointerup notificationclick reset submit touchend"
 * [1] http://kb.mozillazine.org/Dom.popup_allowed_events ***/
user_pref("dom.popup_allowed_events", "click dblclick");
```

**Comment by earthlng** [comment]
```js
/* 2202: prevent websites from disabling new window features
 * [1] http://kb.mozillazine.org/Prevent_websites_from_disabling_new_window_features ***/
user_pref("dom.disable_window_open_feature.close", true);
user_pref("dom.disable_window_open_feature.location", true); // default: true
user_pref("dom.disable_window_open_feature.menubar", true);
user_pref("dom.disable_window_open_feature.minimizable", true);
user_pref("dom.disable_window_open_feature.personalbar", true); // bookmarks toolbar
user_pref("dom.disable_window_open_feature.resizable", true); // default: true
user_pref("dom.disable_window_open_feature.status", true); // status bar - default: true
user_pref("dom.disable_window_open_feature.titlebar", true);
user_pref("dom.disable_window_open_feature.toolbar", true);
/* 2203: disable meddling with open windows ***/
user_pref("dom.allow_scripts_to_close_windows", false); // default: false
user_pref("dom.disable_window_flip", true); // window z-order - default: true
user_pref("dom.disable_window_move_resize", true);
```
`dom.allow_scripts_to_close_windows` is default false + `dom.disable_window_flip` default true, even in ESR52.x, so I think we can remove them. allow_scripts_to_close_windows probably only still exists for some of their tests. I'll address your other questions and points later.

**Comment by Thorin-Oakenpants** [comment]
status bar .. there isn't one any more :). I'd like to keep the whole lot together as a set to be complete, rather than drop a couple off just because of their default values.

**Comment by Thorin-Oakenpants** [comment]
OK, so moved out the context menu and leave this page, renamed section, redone 2202+2203 as suggested, its a bit clearer

Notes
- I kept the same numbers 2202+2203 (we have some in deprecated, I checked and they still fit the new sections) - unless you want to shuffle everything down one number :question: 
- `2204`+ `2205` shuffled order as u said
- `2205` - the ESR needs `3` bit .. need to retest, but stable still needs this to be `not-2`, so I would reword this to be correct
- `2707` `dom.disable_open_during_load` = new to js
- `2208 + 2209` = were `2415 + 2415b` respectively

```js
/* 2204: open new windows in a new tab instead
 * 1=current window, 2=new window, 3=most recent window
 * [SETTING] Options>General>Tabs>Open new windows in a new tab instead ***/
user_pref("browser.link.open_newwindow", 3);
/* 2205: disable links opening in a new window
 * This is to stop malicious window sizes and screen res leaks etc in conjunction with
 * 2203's dom.disable_window_move_resize=true and 2206's full-screen-api.enabled=false
 * [NOTE] ESR52 and lower requires 2204's browser.link.open_newwindow=3
 * [NOTE] RFP (4500) already resizes new windows to cover screen res leaks
 * [NOTE] You can still right click a link and select open in a new window
 * [TEST] https://people.torproject.org/~gk/misc/entire_desktop.html
 * [1] https://trac.torproject.org/projects/tor/ticket/9881 ***/
user_pref("browser.link.open_newwindow.restriction", 0);
/* 2206: disable Fullscreen API [SETUP]
 * [NOTE] You can still manually toggle the browser's fullscreen state (F11), 
 * but this pref will disable embedded video/game fullscreen controls, e.g. youtube
 * [TEST] https://developer.mozilla.org/samples/domref/fullscreen.html ***/
user_pref("full-screen-api.enabled", false);
/* 2207: block popup windows
 * [SETTING] Options>Privacy & Security>Permissions>Block pop-up windows ***/
user_pref("dom.disable_open_during_load", true);
/* 2208 set max popups from a single non-click event - default is 20! ***/
user_pref("dom.popup_maximum", 3);
/* 2209: limit events that can cause a popup
 * default is "change click dblclick mouseup pointerup notificationclick reset submit touchend"
 * [1] http://kb.mozillazine.org/Dom.popup_allowed_events ***/
user_pref("dom.popup_allowed_events", "click dblclick");
```

**Comment by Thorin-Oakenpants** [comment]
Test page: https://people.torproject.org/~gk/misc/entire_desktop.html
Settings: `browser.link.open_newwindow.restriction`=`0` + RFP is enabled
Pref being tested: `browser.link.open_newwindow`

STABLE
- value 1 = about:blank fails to open, returns current res (inner window)
- value 2 = about:blank opens in a new window, returns RFP new window results
  * in my case it says 1400x800, but if I test the new window res at browserspy I get 1400x799
  * that's not an issue, as long as all windows are doing that (i.e height being out by x pixels) - [1418537](https://bugzilla.mozilla.org/show_bug.cgi?id=1418537)
- value 3 = about:blank opens in a new tab, returns 10x10 (?might have something to do with canvas which uses a 10x10)

So stable still requires a value of 3. Amended draft above. Removed extra notes, combined info, removed reason for 3 to 2204

**Comment by earthlng** [comment]
nits: https://github.com/ghacksuserjs/ghacks-user.js/pull/386

I still think 2204+2205 should be combined, fe like this:
```js
/* 2204: open links targeting new windows in a new tab instead
 * This stops malicious window sizes and screen resolution leaks.
 * You can still right click a link and open in a new window (or middle-click) without any risk.
 * [TEST] https://people.torproject.org/~gk/misc/entire_desktop.html
 * [1] https://trac.torproject.org/projects/tor/ticket/9881 ***/
user_pref("browser.link.open_newwindow", 3);
user_pref("browser.link.open_newwindow.restriction", 0);
```

no need to mention 2203's dom.disable_window_move_resize because JS can only resize a window if the window was opened with window.open and has only 1 tab, which is impossible with these 2 prefs.
no need to mention 2204 because it's now one and the same, and no need to mention 2206 because that needs to be disabled independently of this anyway.

**Comment by Thorin-Oakenpants** [comment]
I'm OK with that, just do it :+1:  .. **except** .. 

> You can still right click a link and open in a new window (or middle-click) without any risk

`without any risk`, I would drop those three words. Its too open ended and I don't want to give the wrong impression. There is a risk in some configurations, although, if they already have a unique window size (manually resizing, not using RFP, or maximized) then it simply redoes that same window size (not sure what happens when in full screen), so yeah, no more risk that already exists. I'd just feel better if that was left out. Up to you

**Comment by Thorin-Oakenpants** [comment]
^^ does that mean you will also renumber 2206 to 2205?

**Comment by earthlng** [comment]
https://github.com/ghacksuserjs/ghacks-user.js/pull/387

**Comment by Thorin-Oakenpants** [comment]
done: note the extra changes I made
- used the word `enforce` (we have ten of those) instead of `open` (we have none of those)
- added the RFP new window info note back in
- added inline comment `// 1=current, 2=new, 3=most recent` (good info to keep)

**Comment by earthlng** [comment]
don't like any of the extra changes tbh. Why do you always have to come in and make things worse? ;)

- open is much better and 2204 was "open new windows in a new tab instead" same as in the UI and you didn't mind then either
- who cares what RFP does to new windows when the whole purpose of 2203 is to stop new windows?! adds nothing but confusion IMO. the only thing RFP actually does here is make things worse because of the 10x10 thingy
- I left that out on purpose because it **has** to be 3. I didn't add `// default: 3` either because that also could encourage people to comment it out

**Comment by Thorin-Oakenpants** [comment]
> someone :) should probably report that 10x10 issue to mozilla

I have, to Arthur and Tom Ritter

**Comment by Thorin-Oakenpants** [comment]
IDK if it's worse, that's subjective

* RFP info - I can take it out if you REALLY want me to. As for the whole purpose being to stop new windows, we already added "You can still right-click a link and open in a new window (or middle-click).", so I don't think that argument holds much weight. If we tell em they can still open a new window, then at least let them know with RFP they are covered
> me:  RFP covers new window sizes
> you:  oh I didn't know that

* I wanted the values info because in future, I may want to know what they are meant to do, but I get your point (personally I think its better to explicitly say it must be 3). I spose it's easy enough to re-look it up - I'll take em out, but if anyone comes asking about issues to do with it, I'll refer them to you :trollface:  

* open vs enforce .. its just a word man .. let it go :frozen:

**Comment by earthlng** [comment]
> at least let them know with RFP they are covered

similar to "without any risk". With or without RFP -> "no more risk than already exists"

**Comment by Thorin-Oakenpants** [comment]
Stop picking on me. Don't hate me because I'm beautiful. I undid it all.

I am a firm believer in more information is better - with the RFP line it informed users of the pref that was all. And we told them that its bypassed (common knowledge, context menus etc), so due diligence. **Not a big deal*** - RFP users know, and we flipped that as active. So screw it (see your own comment "I didn't know that"). I'll send any complaints & inquiries your way :grin: 

Still going to gloat one day when someone asks why links won't open :trollface: because someone changed the value to 2 .. but knowing my luck it'll never happen .. you win

**Comment by earthlng** [comment]
links will open just fine with value 2 BUT in new windows instead of tabs :)
Okay I'll stop picking on you - at least for today :trollface: 

> Don't hate me because I'm beautiful

pic or it didn't happen :)

> I undid it all

wooohooo! I did it - I finally had the last word for once - woooohooo! :smile:

**Comment by Thorin-Oakenpants** [comment]
https://github.com/ghacksuserjs/ghacks-user.js/issues/371#issuecomment-376065691

> value 1 = about:blank ***fails to open***, returns current res (inner window)

Soz, I meant value 1
